server/src/main/java/org/elasticsearch/search/aggregations/metrics/TDigestState.java

```diff
@@ -78,12 +87,14 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
-        int h = getClass().hashCode();
```

Comment:
I had to remove this to make `equals` and `hashCode` work. I am not sure why we had to use the class name here...

Reply:
I think in this case it isn't important to take classname into account when generating hashcode. So this is ok.

In theory if there are two empty states one with class `TDigestState` and one another class then these would yield the different hashcodes, but for equals that didn't matter because it only checks whether the other class is an instance of `TDigestState` (which they both are). With this change they generate that same hashcode.